src/app/Http/Controllers/Auth/MyAccountController.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Backpack\Base\app\Http\Controllers\Auth;
+
+use Auth;
+use Backpack\Base\app\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Validation\Rule;
+use Validator;
+
+class MyAccountController extends Controller
+{
+    protected $data = [];
+
+    public function __construct()
+    {
+        $this->middleware('admin');
+    }
+
+    public function showAccountInfoForm()
+    {
+        $this->data['title'] = trans('backpack::base.my_account');
+        $this->data['user'] = Auth::user();
+
+        return view('backpack::auth.account.update_info', $this->data);
+    }
+
+    public function showChangePasswordForm()
+    {
+        $this->data['title'] = trans('backpack::base.my_account');
+        $this->data['user'] = Auth::user();
+
+        return view('backpack::auth.account.change_password', $this->data);
+    }
+
+    public function update(Request $request)
+    {
+        $user = Auth::user();
+
+        $validator = Validator::make($request->all(), [
```

Comment:
I think maybe we should extract the validation logic away from the saving/processing of data - its likely users will want to add an extra field - it seems a shame for them to have to duplicate all this extra stuff?

Reply:
Domn. You're right, this isn't as clean as it could be. I've just split the validation into FormRequests, so it's squeaky-clean now :D